### ISSUE TITLE
Arduino IDE

### ISSUE THREAD

**Issue by oleglyan** [jgeisler0303/AGCON#2]
Hello

Thanks for the hard work on this project

So is it possible to work with library for an Arduino which is using CANopen and create custom fully functional object dictionary, SDOs and PDOs, also read and write SDOs and apply EDS via CAN analyzer software?

**Comment by jgeisler0303** [comment]
Hi,
I'm not quite sure, what you mean. If you are looking for a dynamic PDO mapping at run time as opposed to the currently fixed at compile-time mapping, I'm afraid I have to disappoint you. The little atmega is already at the limit of its resources. A dynamic OD would not fit in any more.  In that case you should go with a bigger ARM base CPU and try to get https://github.com/CANopenNode/CANopenNode running. Otherwise, if you are happy with fixed at compile time OD and mapping, https://github.com/jgeisler0303/CANFestivino, the successor of AGCON, can handle up to about 15 OD entries complete with PDO and SDO access.
If your are looking for a CANopen implementation that can load an OD from EDS at run time, maybe http://ortcan.sourceforge.net/ can help you.

**Comment by oleglyan** [comment]
Hi,

I am just curious. I worked with CANopen devices controllers and inverters for EV CityBus.
For experimental purpose applied the [Watterrot CANdiy shield](http://www.watterott.com/en/Arduino-CANdiy-Shield) with Arduino MEGA for EV Touchscreen HMI as [Gateway ](http://evtouchscreen.blogspot.lt/) for Android Tablets and Vehicle CAN, but only hardware messages were implemented.

This was the topic at [Arduino forums](http://forum.arduino.cc/index.php?topic=169218.18) about CAN from my side, so I could implement full functionality. 

The funniest part, that the vehicle still works with Arduino MEGA on-board without any problem.

Could the external memory like SD card or memory chip extend the object dictionary?

Not working with this task at the moment, but bought some modules with CAN chip

**Comment by jgeisler0303** [comment]
No, an SD card would not help. It is the RAM limitation that makes a larger OD impossible.
I did think about persisting the available OD in EEPROM though but never got around.
I don't actively develop this currently since it serves my purposes sufficiently well and my time is limited.

**Comment by oleglyan** [comment]
OK

I understand

But thank you anyway for your big job and support for the public enthusiasts

ср, 15 июн. 2016 г. в 18:44, Jens Geisler <notifications@github.com>:
